### DISCUSSION TITLE
Unique Share Code

### DISCUSSION THREAD

**Discussion by Anashel-RPG** [vladmandic/sdnext#1397] [Ideas]
There is a ChatGPT feature that allows you to link your entire conversation for anyone to access and use. I would be willing to pay for a feature like that in Vlad1111. 

I train a model called RPG v5.0, and I often get asked for the prompt details. Most of the images have metadata. This project has implemented it very well: [](https://github.com/SupaGruen/StableDiffusion-CheatSheet) - You can drag and drop the image and get a clean view of all the metadata. 

If we could share all the setup in a code block or drag an image to automatically import the data into the text-to-image interface, that would be amazing. For example, we could drag the image onto the prompt field or paste the block into the prompt field, and when you click "generate," it would unpack and set the metadata.

Currently you can drag an image in the prompt but it put everything and does not parse it. So, I mean, the drag and drop already work. :)

**This meta data compress looks like this:**
-START-
eNptU01P4zAQvVfqf5gjiAalpV/sLaIButtC1CAOe1m5ySTx1vFEttMSfv2O+4GEtDlY9vjNmzdv4mTzuk7e+j1HxhJQAQKqrkGToxNSYQ5XezQdkMqhRv1jeDu6BqnBVQjWdQpPKTnyJVNAYaiGShhOacg5NCB0zgBDZYtwkJ/C5JBzKGOQyBgwgFyYHWSkfHqrFOylbYUCTXtUA7jKSP9tjdQl1KKUGd+gRlN2rOXuesBKqC0rx5mAH41BayXpAWRSYy2czMBW5IA0ZELzOsmhVc6wIhRK2iNgJzkhdbhHeEadIzuh4VFspdCQoMFPH+aOIos6kzSAnxysO1gLrXkxGaSyJo3OSSiEdsJ2X1orMoYbE64m21Scxx1VsqxUB18Wo95LQ9pbeG6qUK3MoWHlVBrRVN0ASnJVRj7ftDmppoItKmFYPDeryCIE0Da8F8ZXVZYYOt9xn5ZU69iUfq/fe4mforflewzJee5eC5uv2RLrBsCMJUKFIuc9secas915u6WcdfCED2xXLm0hy9YgA9tSdR5zCLjYABZYkKn9xVbw7LVwVHffMxoi4x0w4qDZsoyl1q0T7ji6486j8MMPSsl6e6nxLa/iX4vL1ZJHzvJOuEIR05yP9liVu+MumPESq4U6KbxQ/L/TU0kmKFvrKb+XP+qU6qS0FrrkSXqH07c4SWE24V20TlbxBhbJ+uYGRik/g1/8MoTt9x4enyB9iFYxjG89Mo4XMBqHs+n9OAz5vPwdw2Q4+phN5/3e+nURr+A5Sp9BDLN5MboPp9tLOBwGm+Qp2E8C6czd/Z9ROPMM71EMe4FB4YLaYjBn2jAM+EUEjWm1V/qwWiaQ/uJlyPB4ky5fX2A2xcn4/q7fS6JNytKjIX/Q8G+G5h+b6WLr
-END-

If it was possible to just share that block of text and cut and paste it in Vlad111 UI so it automatically populate everything, it would be awesome.

**The URL encode also could be nice for the one of us that host it.** 
http://XX.XXX.XX.XXX:XXXX/eNptU01P4zAQvVfqf5gjiAalpV%2FsLaIButtC1CAOe1m5ySTx1vFEttMSfv2O%2B4GEtDlY9vjNmzdv4mTzuk7e%2Bj1HxhJQAQKqrkGToxNSYQ5XezQdkMqhRv1jeDu6BqnBVQjWdQpPKTnyJVNAYaiGShhOacg5NCB0zgBDZYtwkJ%2FC5JBzKGOQyBgwgFyYHWSkfHqrFOylbYUCTXtUA7jKSP9tjdQl1KKUGd%2BgRlN2rOXuesBKqC0rx5mAH41BayXpAWRSYy2czMBW5IA0ZELzOsmhVc6wIhRK2iNgJzkhdbhHeEadIzuh4VFspdCQoMFPH%2BaOIos6kzSAnxysO1gLrXkxGaSyJo3OSSiEdsJ2X1orMoYbE64m21Scxx1VsqxUB18Wo95LQ9pbeG6qUK3MoWHlVBrRVN0ASnJVRj7ftDmppoItKmFYPDeryCIE0Da8F8ZXVZYYOt9xn5ZU69iUfq%2Ffe4mforflewzJee5eC5uv2RLrBsCMJUKFIuc9secas915u6WcdfCED2xXLm0hy9YgA9tSdR5zCLjYABZYkKn9xVbw7LVwVHffMxoi4x0w4qDZsoyl1q0T7ji6486j8MMPSsl6e6nxLa%2FiX4vL1ZJHzvJOuEIR05yP9liVu%2BMumPESq4U6KbxQ%2FL%2FTU0kmKFvrKb%2BXP%2BqU6qS0FrrkSXqH07c4SWE24V20TlbxBhbJ%2BuYGRik%2Fg1%2F8MoTt9x4enyB9iFYxjG89Mo4XMBqHs%2Bn9OAz5vPwdw2Q4%2BphN5%2F3e%2BnURr%2BA5Sp9BDLN5MboPp9tLOBwGm%2BQp2E8C6czd%2FZ9ROPMM71EMe4FB4YLaYjBn2jAM%2BEUEjWm1V%2FqwWiaQ%2FuJlyPB4ky5fX2A2xcn4%2Fq7fS6JNytKjIX%2FQ8G%2BG5h%2Bb6WLr

Cheers!

**Comment by brknsoul** [comment]
> automatically import the data into the text-to-image interface

If the image has metadata, one could simply download said image and drag the image into the Process Image tab then click "🠲 text".

![image](https://github.com/vladmandic/automatic/assets/1969381/49cc7dd2-7bae-4bf3-8b4f-1a399ea3f92d)

results in

![image](https://github.com/vladmandic/automatic/assets/1969381/d8d68631-96ce-4ece-aab5-f8190340cabd)

**Comment by vladmandic** [comment]
you can also drop in prompt like you say and just follow that with one more click - `read generation parameters` - to actually parse the metadata.

**Reply by brknsoul** [reply]
I don't see a "read generation parameters" button...

**Reply by vladmandic** [reply]
first small button under generate/skip/pause/interrupt

**Reply by brknsoul** [reply]
Oh, so drop an image to Process, click "🠲 text", then click 📕?

**Reply by vladmandic** [reply]
you can either drop image to process and click ->text OR you can drop image to prompt directly and click 📕. both work.

**Reply by brknsoul** [reply]
Oh! Nifty! TIL!! ;-)

**Reply by Anashel-RPG** [reply]
Thanks! Drag / Dropping the image and hitting the parsing button is perfect! Do you know if there is a way to automatically add the meta to a upscale image? It seem to get lost when it's upscale. (Or I have a bug)

**Reply by vladmandic** [reply]
if you generate image -> send to process -> upscale -> save, that image will have full metadata **if** generated image was saved. if image was **not** saved, then what is transferred to upscale is just pixel data, there is no metadata to transfer. a1111 does not maintain metadata state outside of image file when transferring between different operations. i've tried to expand that to cover additional scenarios, but this would require a massive rewrite.

**Reply by brknsoul** [reply]
In Settings > Postprocessing, you can add Upscale to both fields. This will enable the Upscaling option in the From Text tab, which will automatically Upscale upon gen completion.

![image](https://github.com/vladmandic/automatic/assets/1969381/473b15fb-d9dd-42e9-ae15-71568440652f)

**Comment by Aptronymist** [comment]
> you can also drop in prompt like you say and just follow that with one more click - `read generation parameters` - to actually parse the metadata.

Would it be worth it to add that `read generation parameters` function to the generate image button, in addition, as suggested though?
There's not really an intuitive way to have people understand the use of that button. 
Let's say if the field is full of the full image data, it won't actually generate, it will instead just process as per book button?

Now that I think of it, perhaps add in the ability to hit the book button and it will (if prompt is empty) directly copy generation data from your clipboard buffer, discarding if it's not valid so you don't get other data, and be ready to generate? 
That makes the button a lot more useful than it currently is. 

That seems like a few easy QOL changes that would make sure this wouldn't even really be a question again, and give a nice boost to usability.
